### PR TITLE
StepicClient now uses OAuth2

### LINKED ISSUE #1: Rewrite StepicClient with OAuth

### PR DESCRIPTION
**Pull request by rev112**
Fixes #1.

### CODE REVIEW

File: games/utils/stepicclient.py [hunk 1]
```diff
@@ -24,69 +19,31 @@ class StepicError(Exception):
...
-
-    def get_attempt(self, id):
-        if not self._is_logged_in:
-            self._login()
-        res = self._request('GET', STEPIC_API_ATTEMPT_URL.format(id=id))
-        if res.status_code == 404:
+    def __init__(self, client_id, client_secret):
+        # Get a token
+        auth = requests.auth.HTTPBasicAuth(client_id, client_secret)
+        response = requests.post('https://stepic.org/oauth2/token/',
```
Comment (psviderski) [inline games/utils/stepicclient.py]: If we started to use constants for such urls in the beginning of this file it's better to have them all at one place.

File: games/utils/stepicclient.py [hunk 2]
```diff
@@ -24,69 +19,31 @@ class StepicError(Exception):
...
+                                 data={'grant_type': 'client_credentials'},
+                                 auth=auth)
+        self.token = json.loads(response.text)['access_token']
```
Comment (psviderski) [inline games/utils/stepicclient.py]: A `Response` object has the `json()` method that makes this json tricks.

File: games/utils/stepicclient.py [hunk 3]
```diff
@@ -24,69 +19,31 @@ class StepicError(Exception):
...
+        self.auth_headers = {'Authorization': 'Bearer ' + self.token}
```
Comment (psviderski) [inline games/utils/stepicclient.py]: Sometimes it's handy to use a requests session and set all the required headers and auth settings on it: http://docs.python-requests.org/en/latest/user/advanced/#session-objects
Then you just use your session with configured headers and don't worry about setting them on each request.
There was a similar behavior already, no idea why you have abandoned that.

**Comment by rev112** [comment]
Updated.

**Comment by psviderski** [comment]
LGTM

**Pull request merged**